### PR TITLE
Allow version validation to ignore NiFi or Registry when checking for parameter support

### LINKED ISSUE #301: Registry import_flow_version calls nifi-api, cannot use library with registry only.
* Nipyapi version: 0.18.0
* NiFi version: 1.14.0
* NiFi-Registry version: 0.8.0
* Python version: 30.10.2
* Operating System: Arch Linux

### Description

I would like to be able to use this library with Nifi Registry only. Registry only operations are also sending nifi-api requests.

### What I Did

When using `nipyapi.versioning.import_flow_version`, the code also calls nifi-api : `/nifi-api/system-diagnostics`.
This comes from validate_parameters_versioning_support function which connects to nifi-api & nifi-registry whatever the command called.

### Urgency

In our environment nifi and nifi-registry is not deployed on the same host, thus needing for the script to know both server instead of using localhost.

### PR DESCRIPTION
**Pull request by ChrisSamo632**
Closes #301

### CODE REVIEW

File: nipyapi/utils.py [hunk 1]
```diff
@@ -530,11 +530,16 @@ def check_version(base, comparator=None, service='nifi'):
...
-def validate_parameters_versioning_support():
-    """Convenience method to check if Parameters are supported"""
-    nifi_check = enforce_min_ver('1.10', bool_response=True)
+def validate_parameters_versioning_support(verify_nifi=True, verify_registry=True):
+    """
+    Convenience method to check if Parameters are supported
+    Args:
+        verify_nifi (bool): If True, check NiFi meets the min version
+        verify_registry (bool): If True, check Registry meets the min version
+    """
```
Comment (ottobackwards) [inline nipyapi/utils.py:539]: can you change the if else structure to match the rest of the code?
if X 
   do
else

Comment (ChrisSamo632) [reply nipyapi/utils.py:539]: 👍

**Comment by ottobackwards** [comment]
Is there a way to add a test for this?

**Comment by ChrisSamo632** [comment]
> Is there a way to add a test for this?

Not that I could see easily because the methods it's calling actually throw an error if the versions don't match and I couldn't see a way of changing the service version as part of a test either (and writing integration tests that need puerile to change

**Comment by Chaffelson** [comment]
Thanks for the fix Chris, and the review Otto. I'll try and get this tested
and merged by Monday

**Comment by Chaffelson** [comment]
I think modifying utils.check_version to not raise an error, but simply assume registry version 0.2.0 if it cannot retrieve the swagger.json is the safer option. Throwing an error is overkill and my implementation was heavy handed - the user can have the option to proceed or halt if they cannot do the check using your true/false option.

I think check_version like this should work, if you care to examine it against your requirements:
```
def check_version(base, comparator=None, service='nifi', default_version='0.2.0'):
    """
    Compares version base against either version comparator, or the version
    of the currently connected service instance.

    Since NiFi is java, it may return a version with -SNAPSHOT as part of it.
    As such, that will be stripped from either the comparator version or
    the version returned from NiFi

    Args:
        base (str): The base version for the comparison test
        comparator (optional[str]): The version to compare against
        default_version (optional[str]): The version to assume the service is
            if the check cannot be completed
        service (str): The service to test the version against, currently
            only supports NiFi

    Returns (int): -1/0/1 if base is lower/equal/newer than comparator

    """
    assert isinstance(base, six.string_types)
    assert comparator is None or isinstance(comparator, six.string_types)
    assert service in ['nifi', 'registry']
    ver_a = version.parse(base)
    if comparator:
        # if b is set, we compare the passed versions
        comparator = strip_snapshot(comparator)
        ver_b = version.parse(comparator)
    elif service == 'registry':
        try:
            config = nipyapi.config.registry_config
            if config.api_client is None:
                config.api_client = nipyapi.registry.ApiClient()
            reg_swagger_def = config.api_client.call_api(
                resource_path='/swagger/swagger.json',
                method='GET', _preload_content=False,
                auth_settings=['tokenAuth', 'Authorization']
            )
            reg_json = load(reg_swagger_def[0].data)
            ver_b = version.parse(reg_json['info']['version'])
        except nipyapi.registry.rest.ApiException:
            log.warning("Unable to retrieve swagger.json from registry "
                        "to check version, assuming version %s" % default_version)
            ver_b = version.parse(default_version)
    else:
        # Working with NiFi
        ver_b = version.parse(
            strip_snapshot(
                # This call currently only supports NiFi
                nipyapi.system.get_nifi_version_info().ni_fi_version
            )
        )
    if ver_b > ver_a:
        return -1
    if ver_b < ver_a:
        return 1
    return 0
```

**Comment by ChrisSamo632** [comment]
@Chaffelson I think that looks OK

I think my 2nd paragraph above about Errors was more aimed at the `enforce_version` method, but upon re-reading that, I think I was mistaken in my comment - the `bool_response` is set to `True` as part of the `validate...` method meaning the Error isn't raised and the (`validate`) warning is reached

Your change to `check_version` would indeed avoid the nested extra Error being raised, where instead the new response with assumed Registry default version would seem more sensible

Do we want that change as part of this MR or a separate one?

**Pull request merged**